packages/next-urql/src/__tests__/with-urql-client.spec.tsx

```diff
@@ -79,39 +81,34 @@ describe('withUrqlClient', () => {
     };
 
     beforeEach(() => {
-      Component = withUrqlClient(
-        ctx => ({
-          url: 'http://localhost:3000',
-          fetchOptions: {
-            headers: { Authorization: (ctx && ctx.req!.headers!.cookie) || '' },
-          },
-        }),
-        mockMergeExchanges
-      )(MockApp);
+      Component = withUrqlClient((ssrExchange, ctx) => ({
+        url: 'http://localhost:3000',
+        fetchOptions: {
+          headers: { Authorization: (ctx && ctx.req!.headers!.cookie) || '' },
+        },
+        exchanges: [(mockSsrExchange = ssrExchange)],
+      }))(MockApp);
     });
 
     it('should allow a user to access the ctx object from Next on the server', async () => {
       Component.getInitialProps &&
         (await Component.getInitialProps(mockContext));
-      expect(spyInitUrqlClient).toHaveBeenCalledWith(
-        {
-          url: 'http://localhost:3000',
-          fetchOptions: { headers: { Authorization: token } },
-        },
-        mockMergeExchanges
-      );
+      expect(spyInitUrqlClient).toHaveBeenCalledWith({
+        url: 'http://localhost:3000',
+        fetchOptions: { headers: { Authorization: token } },
+        exchanges: [mockSsrExchange],
+      });
     });
   });
 
   describe('with mergeExchanges provided', () => {
     const exchange = jest.fn(() => op => op);
 
     beforeEach(() => {
-      mockMergeExchanges.mockImplementation(() => [exchange] as any[]);
-      Component = withUrqlClient(
-        { url: 'http://localhost:3000' },
-        mockMergeExchanges
-      )(MockApp);
+      Component = withUrqlClient(() => ({
+        url: 'http://localhost:3000',
+        exchanges: [exchange] as any[],
+      }))(MockApp);
     });
 
     it('calls the user-supplied mergeExchanges function', () => {
```

Comment:
It looks like we could fully get rid of this particular test and alter the `it` description below it from:

```
'uses exchanges returned from mergeExchanges'
```

to:

```
'uses exchanges defined in the client config'
```

or something of that nature.